src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -654,18 +663,26 @@ public function iShouldHavePaymentMethodAvailableAsTheChoice(string $paymentMeth
      */
     public function iShouldStillBeOnTheCheckoutAddressingStep(): void
     {
-        Assert::same($this->getCart()['checkoutState'], OrderCheckoutStates::STATE_CART);
+        if ($this->sharedStorage->has('checkout_type') && $this->sharedStorage->get('checkout_type') === 'async') {
+            Assert::same($this->getCheckoutState(), AsynchronousOrderCheckoutStates::STATE_CART);
+        } else {
+            Assert::same($this->getCheckoutState(), OrderCheckoutStates::STATE_CART);
+        }
     }
 
     /**
      * @Then I should be on the checkout payment step
      */
     public function iShouldBeOnTheCheckoutPaymentStep(): void
     {
-        Assert::inArray(
-            $this->getCheckoutState(),
-            [OrderCheckoutStates::STATE_SHIPPING_SELECTED, OrderCheckoutStates::STATE_SHIPPING_SKIPPED]
-        );
+        if ($this->sharedStorage->has('checkout_type') && $this->sharedStorage->get('checkout_type') === 'async') {
+            Assert::same($this->getCheckoutState(), AsynchronousOrderCheckoutStates::STATE_SHIPPING_SELECTED);
```

Comment:
If you want to stick to few conditions, we could make it with early return after Assertion to avoid "else" in code

Reply:
I am strongly against empty returns.

Reply:
I am afraid that we are strongly against `else` statement :D but we really like early returns (even empty :P )

Reply:
That's a good presentation/discussion topic `is else so bad that we use empty return?`

Reply:
it is just a consequence of trying to keep functions flat, simple and reduce if statements. Also, it is a little bit easier to track flow paths (but that's subjective). I may have at least idea for a presentation :D